rocAL/source/loaders/image/node_image_loader_single_shard.cpp

```diff
@@ -52,6 +52,23 @@ void ImageLoaderSingleShardNode::init(unsigned shard_id, unsigned shard_count, u
     reader_cfg.set_external_filemode(external_file_mode);
     reader_cfg.set_index_path(index_path);
     reader_cfg.set_sharding_info(sharding_info);
+
+    std::array<std::string, 23> arg_names = {
```

Comment:
[Non-blocking] Minor wording nits in this init: (1) the first error message still refers to 'ImageLoaderNode' rather than 'ImageLoaderSingleShardNode', and (2) the second message reads 'Shard is should be smaller than shard count' – maybe 'Shard id should be smaller than shard count' to match the check. These do not affect behaviour but will make logs easier to understand.